tests/workbookManager.test.ts

```diff
@@ -1,31 +1,52 @@
-import { sharedStringsXmlMock } from "./mocks";
-
-import JSZip from "jszip";
+import { sharedStringsXmlMock, existingSharedStringsXmlMock } from "./mocks";
 
 describe("Workbook Manager tests", () => {
     const workbookManager = new WorkbookManager() as any;
 
     test("Connection XML attributes contain new query name", async () => {
-        const defaultZipFile = await JSZip.loadAsync(workbookTemplate.SIMPLE_QUERY_WORKBOOK_TEMPLATE, { base64: true });
-        await workbookManager.updateSingleQueryAttributes(defaultZipFile, "newQueryName", true);
-        const connectionsXmlString = await defaultZipFile.file(connectionsXmlPath)?.async("text");
-        const hasQueryNewName = connectionsXmlString?.includes("newQueryName");
-        const hasQuery1 = connectionsXmlString?.includes("Query1");
-        expect(hasQueryNewName).toBeTruthy();
-        expect(hasQuery1).toBeFalsy;
+        const {connectionString} = await workbookManager.updateConnections(`<connections xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main" xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" xmlns:xr16="http://schemas.microsoft.com/office/spreadsheetml/2017/revision16" mc:Ignorable="xr16">
+        <connection id="1" xr16:uid="{86BA784C-6640-4989-A85E-EB4966B9E741}" keepAlive="1" name="Query - Query1" description="Connection to the 'Query1' query in the workbook." type="5" refreshedVersion="7" background="1" saveData="1">
+        <dbPr connection="Provider=Microsoft.Mashup.OleDb.1;Data Source=$Workbook$;Location=Query1;" command="SELECT * FROM [Query1]"/>
+        </connection>
+        </connections>`, "newQueryName", true);
+        expect(connectionString.replace(/ /g, "")).toContain('command="SELECT * FROM [newQueryName]'.replace(/ /g, ""));
+        expect(connectionString.replace(/ /g, "")).toContain('name="Query - newQueryName"'.replace(/ /g, ""));
+        expect(connectionString.replace(/ /g, "")).toContain(`description="Connection to the 'newQueryName' query in the workbook."`.replace(/ /g, ""));
+    })
+
+    test("Connection XML attributes contain refreshOnLoad value", async () => {
+        const {connectionString} = await workbookManager.updateConnections(`<connections xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main" xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" xmlns:xr16="http://schemas.microsoft.com/office/spreadsheetml/2017/revision16" mc:Ignorable="xr16">
+        <connection id="1" xr16:uid="{86BA784C-6640-4989-A85E-EB4966B9E741}" keepAlive="1" name="Query - Query1" description="Connection to the 'Query1' query in the workbook." type="5" refreshedVersion="7" background="1" saveData="1">
+        <dbPr connection="Provider=Microsoft.Mashup.OleDb.1;Data Source=$Workbook$;Location=Query1;" command="SELECT * FROM [Query1]"/></connection></connections>`, "newQueryName", true);
+        expect(connectionString.replace(/ /g, "")).toContain('refreshOnLoad="1"');
```

Comment:
parse the xml and check the refresh on load value? let's take this topic into a follow up PR